File: flutter-app/lib/models/feedback.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'feedback.freezed.dart';
part 'feedback.g.dart';

@freezed
class UserFeedback with _$UserFeedback {
  const factory UserFeedback({
    required String incidentId,
    required String action,
    String? entity,
    String? tactic,
    double? newThreshold,
  }) = _UserFeedback;

  factory UserFeedback.fromJson(Map<String, dynamic> json) =>
      _$UserFeedbackFromJson(json);
}

File: python-ai/models/feedback.py
from typing import Optional, Literal
from pydantic import BaseModel


class Feedback(BaseModel):
    incident_id: str
    action: Literal["suppress", "tune", "confirm"]
    entity: Optional[str] = None
    tactic: Optional[str] = None
    new_threshold: Optional[float] = None
